### PR TITLE
remove ng stuff

### PR DESCRIPTION
**Pull request by k-hamza**
**Changes**
 * remove "-ng" stuff
* correction in listen directive when vhost is disabled 
 - Tested on rhel7

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -26,218 +34,116 @@ Installs the Apache package and starts the service.
...
+``apache.vhosts.vhost``
+--------------------------
 
-``apache.mod_security.rules``
------------------------------
+Configures Apache name-based virtual hosts and creates virtual host directories using data from Pillar.
 
-This state can create symlinks based on basic Core Rules package. (Debian only)
-Or it can distribute a mod_security rule file and place it /etc/modsecurity/
+All necessary data must be provided in the pillar
```
Comment (alxwr) [inline README.rst]: @aboe76 @javierbertoli I'd merge this PR as-is, but do we want to add another PR which implements the TOFS pattern?

- https://github.com/saltstack-formulas/template-formula/pull/19
- https://github.com/saltstack-formulas/template-formula/issues/21
Ctrl+F "TOFS pattern vs. pillar"

File: apache/config.sls [hunk 2]
```diff
@@ -29,112 +80,39 @@ include:
...
-      - pkg: apache
-    - watch_in:
-      - module: apache-restart
-    - require_in:
-      - module: apache-restart
-      - module: apache-reload
-      - service: apache
 
-{{ apache.portsfile }}:
+/etc/httpd/conf.d/welcome.conf:
```
Comment (alxwr) [inline apache/config.sls]: This hardcoded path won't fly with Debian(ish) or FreeBSD.

File: README.rst (hunk 1 above)
Comment (myii) [reply README.rst]: @alxwr Let's finalise the portable TOFS implementation in https://github.com/saltstack-formulas/systemd-formula/pull/27.  This should lead to a drop-in solution using `libtofs.jinja`.

File: apache/config.sls (hunk 2 above)
Comment (k-hamza) [reply apache/config.sls]: yes done ! didn't saw it :) thanks

**Comment by k-hamza** [comment]
@aboe76 still some work to do, but at the end it shoudn't have deprecations (if i didn't miss something)

for now it supports only rhel7/httpd 2.4

to support other systems :
- rhel / httpd 2.2 : must add file apache/defaults/Redhat/defaults-apache-2.2.yaml
- Debian : 
	- add file apache/defaults/Debian/defaults-apache-2.2.yaml
	- add file apache/defaults/Debian/defaults-apache-2.4.yaml
	- adapt apache/conf.sls to manage specific conf files to this Os
- Suse : 
	- add file apache/defaults/Suse/defaults-apache-2.2.yaml
	- add file apache/defaults/Suse/defaults-apache-2.4.yaml
	- adapt apache/conf.sls to manage specific conf files to this Os

... same for other systems
	
i think there is a discussion to have about config.sls : 
may be split it this way :
config.sls contains common code and at the end include config-{{ grains['os_family'] }} that will manage specific system conf files :
config-RedHat.sls
config-Debian.sls
config-Suse.sls
...


And about init.sls :
put the content of init.sls in a new file pkg.sls or install.sls
init.sls may contains only some includes

should these adaptations be added in this PR or in another after the merge of this one ?
i am not familiar with other systems, so it may take me some time to achieve that. if you can help i wouldn't say no :-)

**Comment by aboe76** [comment]
@k-hamza I have thought about it, sorry it took some time:

Would love to see suppport for other OSes in a different PR,

Splitting the config.sls, you could create a folder ``config`` 
in which you can have init.sls (for common stuff) and include 
RedHat.sls, Suse.sls, Debian.sls for each OS.

Leave the main init.sls as is for now maybe in a later PR this can be refactored.

By the way, I'm merging the current PR in the develop branch.

**Comment by k-hamza** [comment]
ok for `config` folder, (need some time to learn about the specificities of other systems)
And thanks for the merge !
